packages/gatsby/src/schema/types/node-interface.js

```diff
@@ -36,7 +37,7 @@ const getOrCreateNodeInterface = schemaComposer => {
           )
         },
       },
-      internal: `Internal`,
+      internal: `Internal!`,
```

Comment:
Children should also be [Node!]!. We should return an error if one of the children nodes are null.